internal_public.go

```diff
@@ -148,27 +162,27 @@ func NewActivityTaskHandler(service m.TChanWorkflowService, identity string, log
 func AddWorkflowRegistrationInterceptor(
```

Comment:
I have retained all functions in internal_public separately since they were to be used for catalyst. When the catalyst story is solidified, we can decide to remove this potentially.

On the worker, this is exposed as part of the hostEnv. Does this need to be exposed to users? What is the use case?